test/ex_doc/formatter/html_test.exs

```diff
@@ -38,15 +38,19 @@ defmodule ExDoc.Formatter.HTMLTest do
   test "run generates the html file with the documentation" do
     HTML.run(get_modules, doc_config)
 
-    assert File.regular?("#{output_dir}/CompiledWithDocs.html")
-    assert File.regular?("#{output_dir}/CompiledWithDocs.Nested.html")
+    if doc_config.main do
```

Comment:
@josevalim Any advice to test the `generate_module_page_async` mode function here?